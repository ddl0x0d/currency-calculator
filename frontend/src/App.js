import AppContext from './AppContext';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import Calculator from './components/calculator/Calculator';
import Container from 'react-bootstrap/Container';
import Currencies from './api/Currencies';
import Errors from './components/common/Errors';
import FeeEditor from './components/editor/FeeEditor';
import Navigation from './components/layout/Navigation';
import React, { useEffect, useState } from 'react';

export default function App() {
  const [currencies, setCurrencies] = useState([]);
  useEffect(() => {
    Currencies.get()
      .then(currencies => setCurrencies(currencies))
      .catch(errors => showErrors(errors));
  }, []);

  const [errors, setErrors] = useState([]);
  const showErrors = errors => {
    setErrors(errors);
  };

  return (
    <Router>
      <Navigation />
      <AppContext.Provider value={{ currencies, showErrors }}>
        <Container>
          <Errors messages={errors} />
          <Route path="/" exact component={FeeEditor} />
          <Route path="/editor" component={FeeEditor} />
          <Route path="/calculator" component={Calculator} />
        </Container>
      </AppContext.Provider>
    </Router>
  );
}
