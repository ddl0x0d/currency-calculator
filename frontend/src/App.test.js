import App from './App';
import React from 'react';
import { mount, shallow } from 'enzyme';

describe('<App />', () => {
  it('renders without crashing', () => {
    mount(<App />);
  });

  it('renders correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
