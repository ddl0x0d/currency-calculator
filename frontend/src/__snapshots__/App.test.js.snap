// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<BrowserRouter>
  <Navigation />
  <ContextProvider
    value={
      Object {
        "currencies": Array [],
        "showErrors": [Function],
      }
    }
  >
    <Container
      as="div"
      fluid={false}
    >
      <Errors
        messages={Array []}
      />
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/editor"
      />
      <Route
        component={[Function]}
        path="/calculator"
      />
    </Container>
  </ContextProvider>
</BrowserRouter>
`;
